src/main/java/org/hyperskill/hstest/testing/execution/searcher/CppSearcher.java

```diff
@@ -0,0 +1,24 @@
+package org.hyperskill.hstest.testing.execution.searcher;
+
+import org.hyperskill.hstest.testing.execution.runnable.RunnableFile;
+
+/**
+ * Searches for C++ runnable files
+ * (files with main function)
+ * in the given directory
+ * and returns the first one found.
+ */
+public class CppSearcher extends BaseSearcher {
+    @Override
+    public String extension() {
+        return ".cpp";
+    }
+
+    @Override
+    public RunnableFile search(String whereToSearch) {
+        return simpleSearch(whereToSearch,
+                "int main()",
+                "(^|\n) *int +main\\(.*\\)"
```

Comment:
I saw `(^|\n)\s*int\s+main\s*\(\s*\)` in the Python PR.
Could you use the same template in both PRs?
I like combined one: `(^|\n)\s*int\s+main\s*\(.*\)` because
1. It has `\s` and it is generally more correct than a space.
2. It has `.*` in inside the `( )` since brackets can contain something like `void` or `int argc, char *argv[]` see [examples](https://mr-kumar-abhishek.github.io/blog/2020/04/11/overview-of-main-function-in-c-and-c-plus-plus).